app/src/pages/playground/ModelComboBox.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { Key } from "react-aria-components";
+import {
+  graphql,
+  PreloadedQuery,
+  usePreloadedQuery,
+  useQueryLoader,
+} from "react-relay";
+
+import { ComboBox, ComboBoxItem, ComboBoxProps } from "@phoenix/components";
+
+import modelsQuery, {
+  ModelComboBoxQuery,
+} from "./__generated__/ModelComboBoxQuery.graphql";
+type ModelItem = {
+  name: string;
+  id: string;
+};
+
+type ModelComboBoxProps = {
+  onChange: (model: string) => void;
+  provider: ModelProvider;
+  modelName: string | null;
+  container?: HTMLElement;
+} & Omit<
+  ComboBoxProps<ModelItem>,
+  "children" | "onSelectionChange" | "defaultSelectedKey"
+>;
+
+export function ModelComboBoxLoader({
+  onChange,
+  container,
+  modelName,
+  queryReference,
+  ...comboBoxProps
+}: ModelComboBoxProps & {
+  queryReference: PreloadedQuery<ModelComboBoxQuery>;
+}) {
+  const data = usePreloadedQuery(modelsQuery, queryReference);
+  const items = useMemo((): ModelItem[] => {
+    return data.models.map((model) => ({
+      name: model.name,
+      id: model.name,
+    }));
+  }, [data.models]);
+
+  const [fieldState, setFieldState] = useState({
+    selectedKey: modelName ?? null,
+    inputValue: modelName ?? "",
+  });
+
+  const onSelectionChange = (key: Key | null) => {
+    if (typeof key === "string") {
+      const item = items.find((item) => item.id === key);
+      item?.name != null && onChange(item.name);
+    }
+  };
+
+  const onInputChange = (value: string) => {
+    setFieldState((prevState) => ({
+      inputValue: value,
+      selectedKey: prevState.selectedKey,
+    }));
+  };
+
+  useEffect(() => {
```

Comment:
we can explicitly keep interna state in sync with the model name selection but there are a lot of places where that has to happen so felt like this was easier (also is necessary for trackign when the model gets reset)